packages/rum-core/src/boot/preStartRum.ts

```diff
@@ -38,9 +39,11 @@ export function createPreStartStrategy(
 
   let cachedInitConfiguration: RumInitConfiguration | undefined
   let cachedConfiguration: RumConfiguration | undefined
+  const trackingConsentState = createTrackingConsentState()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
I would have expected to pass the `trackingConsentState` to `doStartRum` to avoid having to recreate the consent state in `startRum`

Reply:
Sure, it's possible but it's yet another value to pass through. Because it doesn't change anything I chose to create it here, but if you prefer I can do what you are suggesting.

Reply:
I would find it easier to follow if we had a single component handling this logic rather than having two components, with one overriding the other.